### PR TITLE
fix(behavior_path_planner): use setter to set current_state_

### PR DESCRIPTION
**Pull request by zulfaqar-azmi-t4**
## Description

The returned type of `updateState` is [discarded](https://github.com/autowarefoundation/autoware.universe/blob/e4fefd4c1312e59d4c2d041a6a8d41b1265ef2b6/planning/behavior_path_planner/include/behavior_path_planner/planner_manager.hpp#L185) in planner_manager,
and if the implementation of `updateState` didn't update the member variable `current_state_`,
the states will not be updated in new architecture.

To solve this issue, the `updateCurrentState` setter is introduced to update the `current_state_` explicitly.

## Related links

None

## Tests performed

1. Compile with OLD and NEW architecture.
2. Testing with CI (OLD) - [TIER IV INTERNAL LINK](https://evaluation.tier4.jp/evaluation/reports/e1c1ac7e-0c3c-569a-9544-89ae76f542bd?project_id=prd_jt)
3. Testing with CI (NEW) - [TIER IV INTERNAL LINK](https://evaluation.tier4.jp/evaluation/reports/7dee208a-e180-5e42-952a-702a78d69361?project_id=prd_jt)

## Notes for reviewers

`current_state_` assignment should be remove.

## Interface changes

None

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by satoshi-ota — APPROVED** [review]
LGTM

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/scene_module_interface.hpp [hunk 1]
```diff
@@ -503,7 +508,7 @@ class SceneModuleInterface
   PlanResult path_candidate_;
   PlanResult path_reference_;
 
-  ModuleStatus current_state_;
+  ModuleStatus current_state_{ModuleStatus::RUNNING};
```
Comment (kosuke55) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/scene_module_interface.hpp:511]: Any reason to set RUNNING here?
The initial value should be `IDLE` for the new architecture and `SUCCESS` for the old one.

https://github.com/autowarefoundation/autoware.universe/blob/main/planning/behavior_path_planner/include/behavior_path_planner/scene_module/scene_module_interface.hpp#L71-L75

Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/include/behavior_path_planner/scene_module/scene_module_interface.hpp:511]: There is no particular reasons.
The value required to be initialized when tick() is called.
Or else, the planning module will not run for old archi. I can try  set and run `IDLE` and `RUNNING` once.

Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/include/behavior_path_planner/scene_module/scene_module_interface.hpp]: fixed in ddd11de07387895cad3a6efe3f700bbb4ba3bb93

**Review by kosuke55 — APPROVED** [review]

**Pull request merged**